### PR TITLE
Add iceberg and csvLoader unittest

### PR DESCRIPTION
**Pull request by qlinsey**
added  iceberg related tests

### CODE REVIEW

**Comment by KimballCai** [comment]
I have checked your code and there are several bugs. I have pushed a new commit to your repo and you can check it. If it is correct, please merge it in your repo.

Besides, please check all your commits and clean useless imports or codes that are not used. And please also check your output log.

**Comment by KimballCai** [comment]
Please check that you can compile the code correctly.

**Comment by qlinsey** [comment]
I removed csv loader related files following PR #39. pushed changes to my remote branch.

**Comment by KimballCai** [comment]
Thanks. This repo runs smoothly on my machine, but please clean the updates that are not used in `cool-core/src/test/java/com/nus/cool/core/model/CoolModelTest.java`.

**Review by KimballCai — CHANGES_REQUESTED** [review]
clean the codes that are not used

File: cool-core/src/test/java/com/nus/cool/core/model/CoolModelTest.java [hunk 1]
```diff
@@ -15,12 +15,15 @@
 import com.nus.cool.model.CoolLoader;
 import com.nus.cool.model.CoolModel;
 import org.testng.annotations.Test;
+import static org.testng.Assert.assertEquals;
```
[the remark this answers is not in the document]
Comment (qlinsey) [reply cool-core/src/test/java/com/nus/cool/core/model/CoolModelTest.java]: removed, thanks!

**Review by qlinsey — COMMENTED** [review]
removed the unused package

**Review by KimballCai — APPROVED** [review]
All test cases pass.

**Comment by qlinsey** [comment]
Should i close this pull request? thanks

**Review by Zrealshadow — COMMENTED** [review]
No meaningful modification in file `CoolModelTest.java`.

for the unittest of IcebergQuery and IcebergSekection, would you mind adding a DataProvider method, defining some static variables in DataProvider and input these as args for unittest method. You can check `HashFieldTest` file. In this layout your unittest will be more scalable, we can add other test data easily.

**Comment by KimballCai** [comment]
Lingze @Zrealshadow has some other comments on your repo about the code format. 

No need to manually close the PR, and we will merge your PR after your modification.

Thanks

**Comment by qlinsey** [comment]
Sure , I will take a look.

**Comment by qlinsey** [comment]
I added DataProvider for two test cases in above, pls check . not sure where else I should add. Thanks.

**Review by Zrealshadow — COMMENTED** [review]
Thanks for your PR. The tests works in my side.
However, there are some improvements.
Firstly, I have to point that these tests is not automatic test. Actually, you should add some assert to compare the result you get and you expect. In your test, you just print these result and judge it manually.
Secondly, I think you should separate the test logic part from test data input part completely for  future scalability. For example, in you `IcebergQueryTest` , you set `@BeforeMethod` annotation in `setUp` method, which means this method will be invoked before every test method. But what you do in `setUp` is just setting some properties for one of your unit test. Obviously, if we add other test method, you have to change this part of code.

**Comment by Zrealshadow** [comment]
> Thanks for your PR. The tests works in my side. However, there are some improvements. Firstly, I have to point that these tests is not automatic test. Actually, you should add some assert to compare the result you get and you expect. In your test, you just print these result and judge it manually. Secondly, I think you should separate the test logic part from test data input part completely for future scalability. For example, in you `IcebergQueryTest` , you set `@BeforeMethod` annotation in `setUp` method, which means this method will be invoked before every test method. But what you do in `setUp` is just setting some properties for one of your unit test. Obviously, if we add other test method, you have to change this part of code.

@KimballCai  No error in these code, I think we can merge it. but it's not automatic test. these code should be refactored and improved in the future.

**Comment by Zrealshadow** [comment]
> I added DataProvider for two test cases in above, pls check . not sure where else I should add. Thanks.

The modification in file `CoolModelTest.java` doesn't make sense. Delete this commit. 
Then, We can merge it.

**Comment by Zrealshadow** [comment]
> > I added DataProvider for two test cases in above, pls check . not sure where else I should add. Thanks.
@qlinsey
The modification in file `CoolModelTest.java` doesn't make sense. Delete this commit.
Then, We can merge it.

**Comment by qlinsey** [comment]
> > I added DataProvider for two test cases in above, pls check . not sure where else I should add. Thanks.
> 
> @qlinsey The modification in file `CoolModelTest.java` doesn't make sense. Delete this commit. Then, We can merge it.



> 
Questions: 
(1) I read CoolModelTest.java, is there any difference from CubeListTest() vs. what I wrote? Does it also require Assert for Array of String?  For IcebergQueryTest, do you suggest me to add AssertEqual to json string? 
(2) I can move IcebergQuery initialization from setUp() to ToStringTest() method, please let me know. thanks.
(3) For CoolModelTest.java, I think that is because I removed the test cases, some more blank lines added. Let me try to remove. Thanks

**Comment by KimballCai** [comment]
> > > I added DataProvider for two test cases in above, pls check . not sure where else I should add. Thanks.
> > 
> > 
> > @qlinsey The modification in file `CoolModelTest.java` doesn't make sense. Delete this commit. Then, We can merge it.
> 
> > 
> 
> Questions: (1) I read CoolModelTest.java, is there any difference from CubeListTest() vs. what I wrote?

please check https://github.com/COOL-cohort/COOL/pull/42/files

> Does it also require Assert for Array of String? For IcebergQueryTest, do you suggest me to add AssertEqual to json string? (2) I can move IcebergQuery initialization from setUp() to ToStringTest() method, please let me know. thanks. (3) For CoolModelTest.java, I think that is because I removed the test cases, some more blank lines added. Let me try to remove. Thanks

**Comment by qlinsey** [comment]
@KimballCai  removed blank lines. thanks

**Comment by KimballCai** [comment]
> > Thanks for your PR. The tests works in my side. However, there are some improvements. Firstly, I have to point that these tests is not automatic test. Actually, you should add some assert to compare the result you get and you expect. In your test, you just print these result and judge it manually. Secondly, I think you should separate the test logic part from test data input part completely for future scalability. For example, in you `IcebergQueryTest` , you set `@BeforeMethod` annotation in `setUp` method, which means this method will be invoked before every test method. But what you do in `setUp` is just setting some properties for one of your unit test. Obviously, if we add other test method, you have to change this part of code.
> 
> @KimballCai No error in these code, I think we can merge it. but it's not automatic test. these code should be refactored and improved in the future.

noted

**Comment by qlinsey** [comment]
thanks, will delete branch.

**Pull request merged**